src/compiler/scala/tools/nsc/CompileClient.scala

```diff
@@ -59,8 +59,7 @@ class StandardCompileClient extends HasCompileSocket with CompileOutputCommon {
 
 object CompileClient extends StandardCompileClient {
   def main(args: Array[String]): Unit = {
-    val ok = try process(args) catch { case _: Exception => false }
-    System.exit(if (ok) 0 else 1)
+    if (!process(args)) throw FatalError("exit(1)")
```

Comment:
@hrhino I think I actually misunderstood your comment at first, and my changes (removing the `try`-`catch`) don't actually help; am I correct about that?

Reply:
Depends what "help" means. I'm a little lost myself on what the best thing to do is, here (since this is a main method the danger is some tool calling it, as I understand). This means they'll get a stack trace in the case of an error that points to the error, or a short `FatalError` trace if `process` returns a `false`, which is arguably better.

The more I think on it, the more I want someone who spoke up on the ticket to speak up here.